docs/source/usage/node_cleaning.rst

```diff
@@ -0,0 +1,259 @@
+
+Automated cleaning is automatically performed before the first workload has been assigned to a node and when hardware is recycled from one workload to another. Automatic cleaning can be customized for to enable or disable certain features of cleaning.
+
+Common use cases for automated cleaning:
+  * Erase disk drives and destroy all the metadata from previous user.
+  * Debug node failures resulting from disk failures by inspecting the automatic cleaning logs in Ironic conductor.
+  * Cut the time spent in manually wiping and cleaning the node after every workload, saving man hours.
+  * Reset raid configuration to a default state after every workload.
+
+Steps involved in automated cleaning:
+  1. Erase devices: out-of-band disk erase operation on all the supported physical drives in the node. Disk erase cannot be performed on physical drives.
+  2. Erase Devices Metadata: Wipe partition tables, signatures, file-system identifications, and other disk devices.
+  3. RAID Delete (Disabled-by-default): Delete any existing RAID configuration. This step needs to be enabled for automated cleaning.
+  4. Setup RAID configuration in advance(Disabled-by-default): Create RAID configuration from a predefined template when the node in enrolled. This step needs to be enabled for automated cleaning.
+
+
+Enable or Disable Automated Cleaning
+------------------------------------
+To enable automated cleaning, ensure that your ironic.conf has following option enabled.
+
+.. prompt:: bash $
+
+  [conductor]
+  automated_clean=true
+
+To disable it, ensure that your ironic.conf has following option disabled.
+
+.. prompt:: bash $
+
+  [conductor]
+  automated_clean=false
+
+Configuring Automated Cleaning Steps
+------------------------------------
+
+Automated cleaning default steps and default priorities:
+  .. list-table::
+    :widths: 25 25 50
+    :header-rows: 1
+    :align: left
+
+    * - Interface
+      - Step
+      - Priority
+    * - [deploy]
+      - erase_devices_priority
+      - 10
+    * - [deploy]
+      - erase_devices_metadata
+      - 99
+    * - [raid]
+      - delete_configuration
+      - 0
+    * - [raid]
+      - create_configuration
+      - 0
+
+
+
+Automated Cleaning extended options available in ironic.conf:
+  .. list-table::
+    :widths: 25 25 50
+    :header-rows: 1
+    :align: left
+
+    * - Interface
+      - Step
+      - Priority
+    * - [deploy]
+      - erase_devices_priority
+      - 10
+    * - [deploy]
+      - erase_devices_metadata
+      - 99
+    * - [`ilo`_]
+      - clean_priority_reset_ilo
+      - 0
+    * - [`ilo`_]
+      - clean_priority_reset_bios_to_default
+      - 10
+    * - [`ilo`_]
+      - clean_priority_reset_secure_boot_keys_to_default
+      - 20
+    * - [`ilo`_]
+      - clean_priority_clear_secure_boot_keys
+      - 0
+    * - [`ilo`_]
+      - clean_priority_reset_ilo_credential
+      - 30
+    * - [`irmc`_]
+      - clean_priority_restore_irmc_bios_config
+      - 0
+
```

Comment:
A question about these steps; if they're non-zero, does that mean they're currently being run? Or are they disabled some other way?

Reply:
Yes if they are non-zero, they are enabled by default, with the stated priority. For some reason, we can not see them in automatic configuration settings.